### PR TITLE
Use os-agnostic helpers

### PR DESCRIPTION
**Pull request by ekohl**

### CODE REVIEW

**Comment by ekohl** [comment]
I should note this is most RFC since I haven't tested it yet. It does make some simplifications that I think should work.

File: fb-install-foreman.bats [hunk 1]
```diff
@@ -42,13 +38,8 @@ setup() {
 
 @test "stop puppet agent (if installed)" {
   tPackageExists "puppet" || skip "Puppet package not installed"
-  if tIsRHEL 6; then
-    service puppet stop; chkconfig puppet off
-  elif tIsFedora; then
-    service puppetagent stop; chkconfig puppetagent off
```
Comment (domcleal) [inline fb-install-foreman.bats]: This disappeared, but it's obsolete anyway, so it's fine.

File: fb-setup-libvirt.bats [hunk 2]
```diff
@@ -9,7 +9,8 @@ load foreman_helper
 @test "install and configure libvirt daemon" {
   if tIsRedHatCompatible; then
     tPackageExists qemu-kvm || yum -y groupinstall Virtualization
-    tPackageExists libvirt || yum -y install libvirt
+    tPackageExists libvirt || tPackageInstall libvirt
+    tServiceStart libvirtd; tServiceEnable libvirtd
     service libvirtd start
     chkconfig libvirtd on
```
Comment (domcleal) [inline fb-setup-libvirt.bats]: These two lines should be removed, they're above now.

File: fb-setup-libvirt.bats [hunk 3]
```diff
@@ -9,7 +9,8 @@ load foreman_helper
...
   elif tIsDebianCompatible; then
```
Comment (domcleal) [inline fb-setup-libvirt.bats]: tPackageInstall for Debian too?

File: os_helper.bash [hunk 4]
```diff
@@ -117,6 +117,50 @@ tPackageVersion() {
...
+      update-rc.d "$1" enable
+    else
+      false # not implemented
+    fi
+  fi
+}
+
+tServiceStart() {
+  if tCommandExists systemctl; then
+    systemctl enable "$1"
```
Comment (domcleal) [inline os_helper.bash]: start

File: fb-install-foreman.bats (hunk 1 above)
Comment (ekohl) [inline fb-install-foreman.bats]: Yes, I checked that and recent Fedora versions ship puppetagent as a symlink to puppet.

**Comment by ekohl** [comment]
Updated

**Comment by domcleal** [comment]
Thanks @ekohl!

**Pull request merged**